cinema/permissions.py

```diff
@@ -0,0 +1,25 @@
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+    def has_permission(self, request, view):
+        return bool(
+            (
+                request.method in SAFE_METHODS
+                and request.user
+                and request.user.is_authenticated
+            )
+            or (request.user and request.user.is_staff)
+        )
```

Comment:
request.user is always there, the only question is if he is authenticated.
Also what is the purpose of bool()? All your checks return boolean, thus you don't need to convert it to bool